modules/UI/UI.js

```diff
@@ -1283,6 +1289,12 @@ UI.showDeviceErrorDialog = function (micError, cameraError) {
                         input.prop("checked");
                 }
             }
+        },
+        null,
+        function () {
+            // Reset dialog reference to null to avoid memory leaks when
+            // user closed the dialog manually.
+            deviceErrorDialog = null;
```

Comment:
I'm not completely comfortable with this unconditional assignment. My idea is that, generally, an instance should clean itself only. Something like:

if (deviceErrorDialog === this)
    deviceErrorDialog = null;